packages/sockethub-platform-xmpp/src/incoming-handlers.js

```diff
@@ -96,28 +82,116 @@ class IncomingHandlers {
     }
   }
 
-  groupBuddy(id, groupBuddy, state, statusText) {
-    this.session.debug('received groupbuddy event: ' + id);
+  // groupBuddy(id, groupBuddy, state, statusText) {
+  //   this.session.debug('received groupbuddy event: ' + id);
+  //   this.session.sendToClient({
+  //     '@type': 'update',
+  //     actor: {
+  //       '@id': `${id}/${groupBuddy}`,
+  //       '@type': 'person',
+  //       displayName: groupBuddy
+  //     },
+  //     target: {
+  //       '@id': id,
+  //       '@type': 'room'
+  //     },
+  //     object: {
+  //       '@type': 'presence',
+  //       status: statusText,
+  //       presence: state
+  //     }
+  //   });
+  // }
+
+  subscribe(to, from, displayName) {
+    this.session.debug('received subscribe request from ' + from);
+    const actor = { '@id': from };
+    if (displayName) {
+      actor.displayName = displayName;
+    }
+    this.session.sendToClient({
+      '@type': "request-friend",
+      actor: actor,
+      target: to
+    });
+  }
+
+  // unsubscribe(from) {
+  //   this.session.debug('received unsubscribe request from ' + from);
+  //   this.session.sendToClient({
+  //     '@type': "remove-friend",
+  //     actor: { '@id': from },
+  //     target: this.session.actor
+  //   });
+  // }
+
+  handleMessage(stanza) {
+    switch (true) {
+      case stanza.attrs.type === 'groupchat':
+        const [ room, actor ] = stanza.attrs.from.split('/');
+        this.notifyGroupChatMessage(room, actor, getMessageBody(stanza));
+        break;
+      case stanza.attrs.type === 'chat':
+        this.notifyChatMessage(stanza.attrs.to, stanza.attrs.from, getMessageBody(stanza));
+        break;
+
+    }
+  }
+
+  handlePresence(stanza) {
+  }
+
+  notifyChatMessage(to, from, message) {
     this.session.sendToClient({
-      '@type': 'update',
+      '@type': 'send',
       actor: {
-        '@id': `${id}/${groupBuddy}`,
         '@type': 'person',
-        displayName: groupBuddy
```

Comment:
Does this mean there is no `displayName` attribute anymore when receiving a message?